### PR TITLE
Added details of rememberDevice option.

### PR DESCRIPTION
**Pull request by srinivasanagandla-okta**
## Description:
- Added information about remember device policy to "Verify factor" section
- Added details of "Remember device policy object" to embedded objects section
- Added rememberDevice attribute to all the factors

## Type of Pull Request:

- [X] Content (Documentation updates or typo-fixes)
- [ ] Blog Post
- [ ] Functional (CSS changes, JS updates, or layout modifications)

### Resolves:
* [OKTA-129449](https://oktainc.atlassian.net/browse/OKTA-129449)

## How Has This Been Tested?
- [X] I have built this locally and verified that it does not break existing functionality.
- [ ] For functional changes, I have tested against Chrome, Firefox, and Safari, and mobile.
- [ ] For functional changes, I have added tests.

## Primary Reviewers:
@nandagopalseshagiri-okta @edjohns-okta @tthompson-okta @mystiberry-okta

### CODE REVIEW

File: _source/_docs/api/resources/authn.md [hunk 1]
```diff
@@ -3220,6 +3220,8 @@ Verifies an enrolled factor for an authentication transaction with the `MFA_REQU
 - [Verify U2F Factor](#verify-u2f-factor)
 - [Verify Call Factor](#verify-call-factor)
 
+> If the sign-on (or app sign-on) [policy](#remember-device-policy-object) allows remembering the device, then end user should be prompted to choose whether current device should be remembered. This would help reduce the number of times user is prompted for MFA on the current device. The user's choice should be passed to Okta using the request parameter `rememberDevice` to the verify endpoint. Default value of `rememberDevice` parameter is `false`.
```
Comment (mystiberry-okta) [inline _source/_docs/api/resources/authn.md]: "the end user"

Comment (mystiberry-okta) [inline _source/_docs/api/resources/authn.md]: "This would help" > "This helps"

File: _source/_docs/api/resources/authn.md [hunk 2]
```diff
@@ -3230,11 +3232,12 @@ Verifies an answer to a `question` factor.
...
------------- | --------------------------------------------------- | ---------- | -------- | -------- |
-fid          | `id` of factor returned from enrollment             | URL        | String   | TRUE     |
-stateToken   | [state token](#state-token) for current transaction | Body       | String   | TRUE     |
-answer       | answer to security question                         | Body       | String   | TRUE     |
+Parameter      | Description                                         | Param Type | DataType | Required |
+-------------- | --------------------------------------------------- | ---------- | -------- | -------- |
+fid            | `id` of factor returned from enrollment             | URL        | String   | TRUE     |
+stateToken     | [state token](#state-token) for current transaction | Body       | String   | TRUE     |
+answer         | answer to security question                         | Body       | String   | TRUE     |
+rememberDevice | user's choice to remember device                    | URL        | Boolean  | FALSE    |
```
Comment (mystiberry-okta) [inline _source/_docs/api/resources/authn.md]: Not quite sure what the description means. "user's choice"?

File: _source/_docs/api/resources/authn.md [hunk 3]
```diff
@@ -3303,11 +3306,12 @@ curl -v -X POST \
...
------------- | --------------------------------------------------- | ---------- | -------- | -------- |
-fid          | `id` of factor                                      | URL        | String   | TRUE     |
-stateToken   | [state token](#state-token) for current transaction | Body       | String   | TRUE     |
-passCode     | OTP sent to device                                  | Body       | String   | FALSE    |
+Parameter      | Description                                         | Param Type | DataType | Required |
+-------------- | --------------------------------------------------- | ---------- | -------- | -------- |
+fid            | `id` of factor                                      | URL        | String   | TRUE     |
+stateToken     | [state token](#state-token) for current transaction | Body       | String   | TRUE     |
+passCode       | OTP sent to device                                  | Body       | String   | FALSE    |
+rememberDevice | user's choice to remember device                    | URL        | Boolean  | FALSE    |
```
Comment (mystiberry-okta) [inline _source/_docs/api/resources/authn.md]: same as line 3240

File: _source/_docs/api/resources/authn.md [hunk 4]
```diff
@@ -3473,11 +3477,12 @@ Verifies an OTP for a `token:software:totp` factor.
...
```
Comment (mystiberry-okta) [inline _source/_docs/api/resources/authn.md]: same as 3240

File: _source/_docs/api/resources/authn.md [hunk 5]
```diff
@@ -3548,10 +3553,11 @@ Sends an asynchronous push notification (challenge) to the device for the user t
...
 
-Parameter    | Description                                         | Param Type | DataType | Required |
------------- | --------------------------------------------------- | ---------- | -------- | -------- |
-fid          | `id` of factor returned from enrollment             | URL        | String   | TRUE     |
-stateToken   | [state token](#state-token) for current transaction | Body       | String   | TRUE     |
+Parameter      | Description                                         | Param Type | DataType | Required |
+-------------- | --------------------------------------------------- | ---------- | -------- | -------- |
+fid            | `id` of factor returned from enrollment             | URL        | String   | TRUE     |
+stateToken     | [state token](#state-token) for current transaction | Body       | String   | TRUE     |
+rememberDevice | user's choice to remember device                    | URL        | Boolean  | FALSE    |
```
Comment (mystiberry-okta) [inline _source/_docs/api/resources/authn.md]: same as 3240

File: _source/_docs/api/resources/authn.md [hunk 6]
```diff
@@ -4001,12 +4007,13 @@ curl -v -X POST \
...
-stateToken   | [state token](#state-token) for current transaction | Body       | String   | TRUE     |
-clientData   | base64 encoded client data from the U2F token       | Body       | String   | TRUE     |
-signatureData| base64 encoded signature data from the U2F token    | Body       | String   | TRUE     |
+Parameter      | Description                                         | Param Type | DataType | Required |
+-------------- | --------------------------------------------------- | ---------- | -------- | -------- |
+fid            | `id` of factor returned from enrollment             | URL        | String   | TRUE     |
+stateToken     | [state token](#state-token) for current transaction | Body       | String   | TRUE     |
+clientData     | base64 encoded client data from the U2F token       | Body       | String   | TRUE     |
+signatureData  | base64 encoded signature data from the U2F token    | Body       | String   | TRUE     |
+rememberDevice | user's choice to remember device                    | URL        | Boolean  | FALSE    |
```
Comment (mystiberry-okta) [inline _source/_docs/api/resources/authn.md]: same as 3240

File: _source/_docs/api/resources/authn.md [hunk 7]
```diff
@@ -4176,11 +4183,12 @@ curl -v -X POST \
...
------------- | --------------------------------------------------- | ---------- | -------- | -------- |
-fid          | `id` of factor                                      | URL        | String   | TRUE     |
-stateToken   | [state token](#state-token) for current transaction | Body       | String   | TRUE     |
-passCode     | OTP sent to device                                  | Body       | String   | FALSE    |
+Parameter      | Description                                         | Param Type | DataType | Required |
+-------------- | --------------------------------------------------- | ---------- | -------- | -------- |
+fid            | `id` of factor                                      | URL        | String   | TRUE     |
+stateToken     | [state token](#state-token) for current transaction | Body       | String   | TRUE     |
+passCode       | OTP sent to device                                  | Body       | String   | FALSE    |
+rememberDevice | user's choice to remember device                    | URL        | Boolean  | FALSE    |
```
Comment (mystiberry-okta) [inline _source/_docs/api/resources/authn.md]: same as 3240

File: _source/_docs/api/resources/authn.md (hunk 2 above)
Comment (mystiberry-okta) [inline _source/_docs/api/resources/authn.md]: Oh, I think I get it. In other docs we say "Indicates whether the user chose to remember the device" or similar. Maybe just "user's decision to remember device" will make it clearer.

File: _source/_docs/api/resources/authn.md (hunk 1 above)
Comment (edjohns-okta) [inline _source/_docs/api/resources/authn.md]: "whether current device" -> "whether the current device"

Comment (edjohns-okta) [inline _source/_docs/api/resources/authn.md]: "times user is prompted" -> "times the user is prompted"

Comment (edjohns-okta) [inline _source/_docs/api/resources/authn.md]: "Default value of 'rememberDevice' parameter is" -> "The default value of the 'rememberDevice' parameter is"

**Comment by srinivasanagandla-okta** [comment]
@edjohns-okta @mystiberry-okta I have updated based on the feedback. Please take a look.

**Pull request merged**